sqlglot/parser.py

```diff
@@ -795,13 +795,19 @@ def _parse_select(self):
             )
 
         if isinstance(this, exp.Subquery) or self._match(TokenType.SELECT):
+            hint=self._parse_hint()
+            _=self._match(TokenType.ALL)
```

Comment:
Should we be matching ALL _or_ DISTINCT?

Reply:
Yes, it is one or the other, I'll try to fix that